### DISCUSSION TITLE
Registering additional endpoints

### DISCUSSION THREAD

**Discussion by nestabentum** [eclipse-emfcloud/emfcloud#231] [ModelServer]
I'm looking for a way to register additional REST Endpoints in my model server - so far I have not found an elegant way to do this. Any ideas or pointers?

**Comment by vhemery** [comment]
That may be a bit outdated, but here are some pointers https://youtu.be/T2r05Ti50Bc?t=1325
I no longer have the code, but that was quite straightforward how I remember it.

Also, make sure you do not add REST endpoints where you could just use custom commands.

**Comment by Pitwng** [comment]
> I'm looking for a way to register additional REST Endpoints in my model server - so far I have not found an elegant way to do this. Any ideas or pointers?

I am attempting to add additional REST endpoints to the Model Server Client implementation.
 Could you provide recommendations or best practices for designing and integrating these endpoints, particularly in the context of EMF-based model management workflows? 
Any guidance on maintaining compatibility with existing APIs or addressing common pitfalls would be greatly appreciated.
And this is the question include the detail and my test:
https://github.com/eclipse-emfcloud/emfcloud/discussions/245
